synapse/lib/lmdbslab.py

```diff
@@ -561,22 +559,23 @@ class Slab(s_base.Base):
     DEFAULT_GROWSIZE = None
 
     @classmethod
-    async def initSyncLoop(clas, inst):
+    def getSlabsInDir(clas, dirn):
+        '''
+        Returns all open slabs under a directory
+        '''
+        toppath = pathlib.Path(dirn).resolve()
+        return [slab for slab in clas.allslabs.values() if toppath in slab.path.parents or toppath == slab.path]
```

Comment:
Seems like a simple prefix search would due here rather than comparing each value in the list of parents